src/rpc/rawtransaction.cpp

```diff
@@ -336,6 +339,120 @@ static UniValue gettxchainlocks(const JSONRPCRequest& request)
     return result_arr;
 }
 
+static UniValue getassetunlockchainlocks(const JSONRPCRequest& request)
+{
+    RPCHelpMan{
+            "getassetunlockchainlocks",
+            "\nReturns the status of given Asset Unlock indexes.\n",
+            {
+                    {"indexes", RPCArg::Type::ARR, RPCArg::Optional::NO, "The Asset Unlock indexes (no more than 100)",
+                     {
+                             {"index", RPCArg::Type::NUM, RPCArg::Optional::OMITTED, "An Asset Unlock index"},
+                     },
+                    },
+            },
+            RPCResult{
+                    RPCResult::Type::ARR, "", "Response is an array with the same size as the input txids",
+                    {
+                            {RPCResult::Type::OBJ, "xxxx", "Asset Unlock index.",
+                             {
+                                     {RPCResult::Type::STR, "", "Status of the Asset Unlock index: {chainlocked|mempooled|null}"},
+                             }},
+                    }
+            },
+            RPCExamples{
+                    HelpExampleCli("getassetunlockchainlocks", "'[\"myindex\",...]'")
+                    + HelpExampleRpc("getassetunlockchainlocks", "[\"myindex\",...]")
+            },
+    }.Check(request);
+
+    const NodeContext& node = EnsureAnyNodeContext(request.context);
+    const CTxMemPool& mempool = EnsureMemPool(node);
+    const LLMQContext& llmq_ctx = EnsureLLMQContext(node);
+    ChainstateManager& chainman = EnsureChainman(node);
+
+    UniValue result_arr(UniValue::VARR);
+    UniValue str_indexes = request.params[0].get_array();
+    if (str_indexes.size() > 100) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Up to 100 indexes only");
+    }
+
+    if (g_txindex) {
+        g_txindex->BlockUntilSyncedToCurrentChain();
+    }
+
+    CBlockIndex* pTipBlockIndex{nullptr};
+    {
+        LOCK(cs_main);
+        pTipBlockIndex = chainman.ActiveChain().Tip();
+    }
```

Comment:
refactor to use WITH_LOCK
```suggestion
    CBlockIndex* pTipBlockIndex= WITH_LOCK(cs_main, return chainman.ActiveChain().Tip());
```